src/algebra/big_operators/fin.lean

```diff
@@ -141,8 +141,36 @@ end fin
 
 namespace list
 
+section monoid
+
+variables [monoid α] {n : ℕ}
+
+/-- Given `f = (a₁, ..., aₙ)` in `αⁿ`, `take_prod f` is `(1, a₁, a₁a₂, ..., a₁...aₙ)` in `αⁿ⁺¹`. -/
+@[to_additive "Given `f = (a₁, ..., aₙ)` in `αⁿ`, `take_sum f` is
+`(1, a₁, a₁ + a₂, ..., a₁ + ... + aₙ)` in `αⁿ⁺¹`."]
+def take_prod (f : fin n → α) (i : fin (n + 1)) : α :=
```

Comment:
It feels a bit weird to have this as a definition when we don't even have a single definition for the product of a tuple.

I wonder if something like `((list.Iio i).map f).prod` should be the preferred spelling in the longer term.
Maybe @YaelDillies has ideas.

(Using that spelling certainly is out of scope for this PR, but inlining this new def or adding a Todo comment might not be)

Reply:
This also seemed a bit weird to me and so I considered inlining. I was on the fence but persuaded myself based on the applications in this PR. I only have a moment now so I'll think again about this later today.

Reply:
In the context of incidence algebras, this is the zeta function (yes, this is related to Riemann's zeta function!). See #11656. I'm not sure about the `fin` packaging however (#11656 gives you `(ℕ → α) → ℕ → α`, not `(fin n → α) → fin (n + 1) → α`).

Reply:
I thought about this some more and I'm more supportive of keeping `take_prod` as it stands (possibly adding a `TODO` if someone has a suggestion).

With regard to the `fin` packaging, I support this also because it caters for an important application. To be specific, the type `(fin n → α) → fin (n + 1) → α)` is exactly what one needs if (as in this PR) one has a (finite) `n`-tuple of elements of `α`.

Of course dissenting views are most welcome!

Reply:
I am not in a position to judge whether we do want such a standalone definition, but I shall point out that `take_prod` is a vague name (what are we taking?). What about `cumul_prod`, `partial_prod`, `accumulate_prod` or something like this?

Reply:
`partial_prod` or something like `tuple.scan_prod` (although I don't know if `tuple` is used anywhre) sounds good to me.

Reply:
Note that we already have [`partial_sups`](https://leanprover-community.github.io/mathlib_docs/order/partial_sups.html#partial_sups)

Reply:
Totally agreed: `take_prod` is no good. I really like `partial_prod`. Now officially proposed.